variable/include/scipp/variable/accumulate.h

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+/// @file Accumulation functions for variables, based on transform.
+/// @author Simon Heybrock
+#pragma once
+
+#include "scipp/variable/shape.h"
+#include "scipp/variable/transform.h"
+
+namespace scipp::variable {
+
+namespace detail {
+template <class... Ts, class Op, class Var, class... Other>
+static void accumulate(const std::tuple<Ts...> &types, Op op,
+                       const std::string_view &name, Var &&var,
+                       Other &&... other) {
+  // Bail out (no threading) if:
+  // - `other` is implicitly broadcast
+  // - `other` are small, to avoid overhead (important for groupby)
+  // - `other` not const, threading for cumulative ops not possible
+  // - reduction to scalar with more than 1 `other`
+  if ((!other.dims().includes(var.dims()) || ...) ||
+      ((other.dims().volume() < 16384) && ...) || // BM_groupby_large_table
+      (!std::is_const_v<std::remove_reference_t<Other>> || ...) ||
+      (sizeof...(other) != 1 && var.dims().ndim() == 0))
+    return in_place<false>::transform_data(types, op, name, var, other...);
+
+  const auto reduce_chunk = [&](auto &&out, const Slice slice) {
+    const bool avoid_false_sharing = out.dims().volume() < 128;
```

Comment:
The three hard coded numbers in this header should be properly explained to aid future maintainers.

Reply:
👍 done

Reply:
If one were to fine tune this that could become useful.